ITEvents/Screens/Favorive tab/FavoritesViewController.swift

```diff
@@ -1,98 +1,73 @@
+    private var events = [EventCollectionCellViewModel]()
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        setUpLayouts()
-        rotationButton.animationDuration = animationDuration
-        registerNibs()
-        
+        setupViewController()
         presenter.setup()
-        
-        setupCollectionView()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        presenter.activate()
     }
     
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         collectionView.isUserInteractionEnabled = true
     }
     
-    func setEvents(_ events: [EventCollectionCellViewModel]) {
-        self.events = events
-        collectionView.reloadData()
-    }
-    
-    func toggleListLayout(to list: Bool) {
-        layoutState = list ? .list : .grid
-        rotationButton.isSelected = list
-    }
-    
-    private func setUpLayouts() {
-        let viewWidth = view.frame.width
-        listLayout = createDisplaySwitcherLayout(forList: true, viewWidth: viewWidth)
-        gridLayout = createDisplaySwitcherLayout(forList: false, viewWidth: viewWidth)
-    }
-    
-    private func setupCollectionView() {
-        let layout = getCurrentLayout()
-        collectionView.collectionViewLayout = layout
+    private func setupViewController() {
+        registerNibs()
+        eventCollectionViewHandler = EventCollectionViewHandler(viewWidth: view.frame.width,
+                                                               collectionView: collectionView,
+                                                               selectedEventAction: selectedEventAction,
+                                                               lastCellWillDisplayAction: lastCellWillDisplayAction)
+        collectionView.delegate = eventCollectionViewHandler
+        collectionView.dataSource = eventCollectionViewHandler
     }
     
     private func registerNibs() {
-        collectionView.register(cellType: EventCollectionViewCell.self)
+        collectionView.register(cellType: ListCollectionViewCell.self)
+        collectionView.register(cellType: GridCollectionViewCell.self)
     }
     
-    @IBAction private func changeLayout(_ sender: Any) {
-        presenter.toggleLayoutState()
-        let layout = getCurrentLayout()
-        let transitionManager = TransitionManager(duration: animationDuration,
-                                                  collectionView: collectionView,
-                                                  destinationLayout: layout,
-                                                  layoutState: layoutState)
-        transitionManager.startInteractiveTransition()
+    func toggleLayout(value isListLayout: Bool) {
+        if eventCollectionViewHandler.isListLayoutSelected == isListLayout { return }
+        eventCollectionViewHandler.toggleListLayout()
```

Comment:
Let's delegate ```isListLayoutSelected``` check to ```eventCollectionViewHandler```.
It will allow us to only have the following code here:
```swift
func toggleLayout(value isListLayout: Bool) {
   eventCollectionViewHandler.toggleListLayout(value: isListLayout)
}
```